File: UCFPT/22locals/local2022FinalRoundData/solutions/pseudo/pseudo.py
# Arup Guha
# 8/9/2022
# Solution to UCF Locals Problem: Pseudo Pseudo Random Numbers

# Returns max run of 0s or 1s in the binary representation of n.
def maxRun(n,bits):

    curBit = -1
    curRun = 0
    maxRun = 0

    # Peel off bit by bit.
    for i in range(bits):

        # Next bit. Clunky sorry.
        bit = 0
        if (n & (1<<i)) != 0:
            bit = 1
        
        # New run.
        if bit != curBit:
            curBit = bit
            curRun = 1

        # Extend old run.
        else:
            curRun += 1

        # Update and go to next bit.
        maxRun = max(maxRun, curRun)

    return maxRun
    
def main():

    # Get input.
    toks = [int(x) for x in input().split()]

    # Store result here.
    res = 0

    # Try each bit string.
    for i in range(1<<toks[0]):
        if maxRun(i, toks[0]) <= toks[1]:
            res += 1

    # Ta da!
    print(res)

# Run it!
main()

File: UCFPT/22locals/local2022FinalRoundData/solutions/pseudo/pseudo.java
// Arup Guha
// 8/9/2022
// Solution to UCF Locals Problem: Pseudo Pseudo Random Numbers

import java.util.*;

public class pseudo {

	public static void main(String[] args) {
	
		// Get input.
		Scanner stdin = new Scanner(System.in);
		int n = stdin.nextInt();
		int k = stdin.nextInt();
		
		int res = 0;
		
		// Go through all choices.
		for (int i=0; i<(1<<n); i++)
			if (maxRun(i, n) <= k)
				res++;
				
		// Ta da!
		System.out.println(res);
	}
	
	// Returns the max run of the same bit in n when represented with bits # of bits.
	public static int maxRun(int n, int bits) {
	
		// Go through each bit.
		int curRun = 0, maxRun = 0, curBit = -1;
		for (int i=0; i<bits; i++) {
		
			// Get this bit.
			int bit = (n & (1<<i)) == 0 ? 0 : 1;
			
			// New streak.
			if (bit != curBit) {
				curRun = 1;
				curBit = bit;
			}
			
			// Extend old streak.
			else
				curRun++;
				
			// Update max if necessary.
			maxRun = Math.max(maxRun, curRun);
		}
		
		return maxRun;
	}
}